### PR TITLE
basic celestrius flow in octoprint

### PR DESCRIPTION
**Pull request by aarontuftss**

### CODE REVIEW

File: octoprint_obico/webcam_capture.py [hunk 1]
```diff
@@ -49,11 +49,16 @@ def webcam_full_url(url):
 
 @backoff.on_exception(backoff.expo, Exception, max_tries=3)
 @backoff.on_predicate(backoff.expo, max_tries=3)
-def capture_jpeg(plugin, force_stream_url=False):
+def capture_jpeg(plugin, force_stream_url=False, use_nozzle_config=False):
     MAX_JPEG_SIZE = 5000000
 
-    webcam_settings = octoprint_webcam_settings(plugin._settings)
-    snapshot_url = webcam_full_url(webcam_settings.get("snapshot", ''))
+    if use_nozzle_config:
```
Comment (kennethjiang) [inline octoprint_obico/webcam_capture.py:55]: `use_nozzle_config` is not an elegant solution. Change `capture_jpeg` to take webcam_config as a parameter instead. Make it closer to the same func in moonraker-obico.

File: octoprint_obico/nozzlecam.py [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+        while True:
+            if not self.nozzle_config: # cut out loop if nozzlecam not set up
+                return
+            if self.on_first_layer == True:
+                try:
+                    self.send_nozzlecam_jpeg(capture_jpeg(self.nozzle_config, use_nozzle_config=True))
+                    _logger.debug('Nozzle cam Jpeg captured & sent')
+                except Exception as e:
+                    _logger.warning('Failed to capture jpeg - ' + str(e))
```
Comment (kennethjiang) [inline octoprint_obico/nozzlecam.py:27]: Error handing is where you need to improve. When handling errors, think about:
1. Who will consume this error message. Users? us? or both?
2. How to deliver this error message to the intended audience?
3. Make sure we have enough info for debug, in this case, the stack trace will be gone.

File: octoprint_obico/nozzlecam.py [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+                data = {'viewing_boost': 'true'}
+                server_request('POST', '/ent/api/nozzle_cam/pic/', self.plugin, timeout=60, files=files, data=data, skip_debug_logging=True, headers=self.plugin.auth_headers())
+            except Exception as e:
+                _logger.warning('Failed to post jpeg - ' + str(e))
+
+    def notify_server_nozzlecam_complete(self):
+        self.on_first_layer = False
+        try:
+            data = {'nozzlecam_status': 'complete'}
+            server_request('POST', '/ent/api/nozzle_cam/first_layer_done/', self.plugin, timeout=60, files={}, data=data, skip_debug_logging=True, headers=self.plugin.auth_headers())
```
Comment (kennethjiang) [inline octoprint_obico/nozzlecam.py:43]: We want debug logging for this call. So we shouldn't skip it.

File: octoprint_obico/nozzlecam.py [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+        while not self.plugin.linked_printer.get('id', None): #main loop waits for auth token -  need to guarantee we have the printerID before calling DB
+            time.sleep(1)
+        self.nozzle_config = self.create_nozzlecam_config()
+
+        while True:
+            if not self.nozzle_config: # cut out loop if nozzlecam not set up
+                return
+            if self.on_first_layer == True:
+                try:
+                    self.send_nozzlecam_jpeg(capture_jpeg(self.nozzle_config, use_nozzle_config=True))
```
Comment (kennethjiang) [inline octoprint_obico/nozzlecam.py:24]: The first arg of capture_jpeg is "plugin". You tried to pass nozzle_config here! You need to think about how you can minimize the chance of making basic mistakes like this.

Comment (aarontuftss) [reply octoprint_obico/nozzlecam.py:24]: self.nozzle_config is correct here. The `use_nozzle_config = True` argument ensures that `capture_jpeg` adapts to & will work with `self.nozzle_config`

Comment (kennethjiang) [reply octoprint_obico/nozzlecam.py:24]: > self.nozzle_config is correct here. The `use_nozzle_config = True` argument ensures that `capture_jpeg` adapts to & will work with `self.nozzle_config`

You shouldn't need `use_nozzle_config = True` if you tried to make `nozzle_config` the same behavior as webcam_config.

File: octoprint_obico/nozzlecam.py [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+                    _logger.debug('Nozzle cam Jpeg captured & sent')
+                except Exception as e:
+                    _logger.warning('Failed to capture jpeg - ' + str(e))
+            time.sleep(0.2)
+
+    def send_nozzlecam_jpeg(self, snapshot):
+        if snapshot:
+            try:
+                files = {'pic': snapshot}
+                data = {'viewing_boost': 'true'}
```
Comment (kennethjiang) [inline octoprint_obico/nozzlecam.py:34]: viewing_boost serves a very special purpose in the original code. it's not applied here. Please check with me when you try to copy the code that you don't understand.

File: octoprint_obico/nozzlecam.py (hunk 3 above)
Comment (kennethjiang) [inline octoprint_obico/nozzlecam.py:43]: `files={}` shouldn't be here.

File: octoprint_obico/nozzlecam.py [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+            data = {'nozzlecam_status': 'complete'}
+            server_request('POST', '/ent/api/nozzle_cam/first_layer_done/', self.plugin, timeout=60, files={}, data=data, skip_debug_logging=True, headers=self.plugin.auth_headers())
+            _logger.debug('server notified 1st layer is done')
+        except Exception as e:
+            _logger.warning('Failed to notify 1st layer completed' + str(e))
+
+    def create_nozzlecam_config(self):
+        try:
+            printer_id = self.plugin.linked_printer.get('id')
+            info = server_request('GET', f'/ent/api/printers/{printer_id}/ext/', self.plugin, timeout=60, files={}, data={}, skip_debug_logging=True, headers=self.plugin.auth_headers())
```
Comment (kennethjiang) [inline octoprint_obico/nozzlecam.py:51]: When you copy code, please make sure you understand each parameter. Most of the time, blindly copying code is not a good ideal unless for the most trivial code.

**Pull request merged**